cartographer/io/fake_file_writer.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_IO_FAKE_FILE_WRITER_H_
+#define CARTOGRAPHER_IO_FAKE_FILE_WRITER_H_
+
+#include <memory>
+#include <string>
+
+#include "cartographer/io/file_writer.h"
+
+namespace cartographer {
+namespace io {
+
+// Fakes a FileWriter by just writing the data to a std::string.
+class FakeFileWriter : public FileWriter {
+ public:
+  FakeFileWriter(const std::string& filename,
+                 std::shared_ptr<std::string> on_close_output =
+                     std::make_shared<std::string>());
+  ~FakeFileWriter() override = default;
+
+  bool WriteHeader(const char* data, size_t len) override;
+  bool Write(const char* data, size_t len) override;
+  bool Close() override;
+  std::string GetFilename() override;
```

Comment:
rename `GetFilename()` to `filename()`. It is allowed for getters/setters to have names like `filename` and `set_filename`.

Reply:
std::string GetFilename() is defined in the FileWriter interface. This would require changes in cartographer_ros. As both naming conventions are allowed, I would like to stick with the current implementation. 
I named 
std::string GetOutput() const;
to be consistent with the GetFilename().